std/container/rbtree.d

```diff
@@ -1494,30 +1525,63 @@ assert(equal(rbt[], [5]));
         return Range(_begin, _firstGreaterEqual(e));
     }
 
-    /**
-     * Get a range from the container with all elements that are == e according
-     * to the less comparator
-     *
-     * Complexity: $(BIGOH log(n))
-     */
-    Range equalRange(Elem e)
+    /// Ditto
+    ConstRange lowerBound(Elem e) const
+    {
+        return ConstRange(_begin, _firstGreaterEqual(e));
+    }
+
+    /// Ditto
+    ImmutableRange lowerBound(Elem e) immutable
+    {
+        return ImmutableRange(_begin, _firstGreaterEqual(e));
+    }
+
+    private auto _equalRange(this This)(Elem e)
     {
+        import std.typecons;
         auto beg = _firstGreaterEqual(e);
         if(beg is _end || _less(e, beg.value))
             // no values are equal
-            return Range(beg, beg);
+            return tuple(beg, beg);
         static if(allowDuplicates)
         {
-            return Range(beg, _firstGreater(e));
+            return tuple(beg, _firstGreater(e));
         }
         else
         {
             // no sense in doing a full search, no duplicates are allowed,
             // so we just get the next node.
-            return Range(beg, beg.next);
+            return tuple(beg, beg.next);
```

Comment:
I think you can redo this function to avoid the additional template machinery below:

```D
auto equalRange(This this)(Elem e)
{
   alias RangeType = RBRange!(typeof(*_begin)*);
   ...
   return RangeType(beg, _firstGreater(e));
   ... // etc
}
```